gobblin-modules/gobblin-http/src/main/java/gobblin/converter/AvroR2JoinConverter.java

```diff
@@ -42,8 +46,19 @@ protected void fillHttpOutputData(Schema schema, GenericRecord outputRecord, Res
 
   @Override
   protected HttpClient<RestRequest, RestResponse> createHttpClient(WorkUnitState workUnitState) {
-    //TODO: should use R2ClientFactory to initilize R2Client
-    return null;
+    Config config = ConfigBuilder.create().loadProps(workUnitState.getProperties(), CONF_PREFIX).build();
+    config = config.withFallback(DEFAULT_FALLBACK);
```

Comment:
We could DRY line 49-50:
1. These `create` methods accept `Config` instead of `WorkUnitState`
2. Apply the `DEFAULT_FALLBACK` in the `init` function
3. On create the HttpClient, pass the broker as the second argument

Since config is namespace driven. This approach helps restrict a subclass to create those constructs based on only the configurations inside that namespace, avoid scoping issues